include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge_sort.h

```diff
@@ -57,6 +59,154 @@ struct __leaf_sort_kernel
     }
 };
 
+template <std::uint16_t __data_per_workitem>
+struct __group_merge_path_sorter
+{
+    template <typename _Storage, typename _Compare, typename _Size1, typename _Size2>
+    bool
+    sort(const sycl::nd_item<1>& __item, _Storage& __storage, _Compare __comp, _Size1 __start, _Size1 __end,
+         _Size2 __sorted, std::uint32_t __workgroup_size) const
+    {
+        const std::uint32_t __sorted_final = __data_per_workitem * __workgroup_size;
+
+        const std::uint32_t __id = __item.get_local_linear_id() * __data_per_workitem;
+
+        bool __data_in_temp = false;
+        std::uint32_t __next_sorted = __sorted * 2;
+        // ctz precisely calculates log2 of an integral value which is a power of 2, while
+        // std::log2 may be prone to rounding errors on some architectures
+        std::int16_t __iters = sycl::ctz(__sorted_final) - sycl::ctz(__sorted);
+        for (std::int16_t __i = 0; __i < __iters; ++__i)
+        {
+            const std::uint32_t __id_local = __id % __next_sorted;
+            // Borders of the ranges to be merged
+            const std::uint32_t __start1 = std::min<std::uint32_t>((__id / __next_sorted) * __next_sorted, __end);
```

Comment:
May you explain, why should we use here `<std::uint32_t>` ?

Reply:
What's about
```C++
const std::uint32_t __start1 = std::min(__id - __id % __next_sorted, __end);
```
?

Reply:
> May you explain, why should we use here <std::uint32_t> ?

It is used to help `std::min` with selection of a common type for the arguments. Perhaps, it is a sign that the function must explicitly take `__end` parameter as `std::uint32_t` to highlight its requirements.

> What's about const std::uint32_t __start1 = std::min(__id - __id % __next_sorted, __end);

That's a good idea, especially because `__id % __next_sorted` part has already been calculated.

Reply:
Done

Reply:
I don't see the change for
https://github.com/oneapi-src/oneDPL/pull/1735/files/d29e35c10604b2f0e55336ab59bba3e67fbcb738#r1704399376

Reply:
It is in [1cd360d](https://github.com/oneapi-src/oneDPL/pull/1735/commits/1cd360dce432a016372508852871ef44cc99d0bf).